src/services/s3/backend.rs

```diff
@@ -457,7 +458,7 @@ impl Builder {
         self.server_side_encryption_customer_algorithm = Some(algorithm.to_string());
         self.server_side_encryption_customer_key = Some(base64::encode(key));
         self.server_side_encryption_customer_key_md5 =
-            Some(base64::encode(md5::compute(key).as_slice()));
+            Some(base64::encode(Md5::digest(key).as_slice()));
```

Comment:
Need to check if the results are the same, I have only checked a few simple examples.